client/components/timezone-dropdown/index.jsx

```diff
@@ -3,50 +3,100 @@
  * External Dependencies
  */
 import React from 'react';
-import noop from 'lodash/noop';
+import i18n from 'lib/mixins/i18n';
 
 /**
  * Internal dependencies
  */
+import wpcom from 'lib/wp';
 import Dropdown from 'components/select-dropdown';
-import i18n from 'lib/mixins/i18n';
 
-export default React.createClass( {
-	displayName: 'TimezoneDropdown',
+/**
+ * Module variables
+ */
+const { Component, PropTypes } = React;
+const noop = () => {};
+const undocumented = wpcom.undocumented();
+
+class TimezoneDropdown extends Component {
+	constructor() {
+		super();
 
-	propTypes: {
-		selectedZone: React.PropTypes.string,
-		onSelect: React.PropTypes.func
-	},
+		// bound methods
+		this.onSelect = this.onSelect.bind( this );
```

Comment:
yes ..., but `onSelect` is called from a component property and there the function isn't bounded and on other hand we should avoid call a bound method from the react prop.

```es6
<TimezoneDropdown
  onClick={ this.onSelect.bind( this ) }
/>
```